src/util/file_reader.js

```diff
@@ -97,7 +97,7 @@ class NewlineReader {
             if (this.lock) await fh.flock(this.fs_context, this.lock);
 
             this.fh = fh;
-        } finally {
+        } catch {
```

Comment:
First - is this valid try-catch syntax? I think it needs to have `catch (err) {...}`.
Second - do we mean to "swallow" this error? shouldn't we rethrow it up to the caller?

Reply:
Yes, the syntax is valid - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch. `exceptionVar` is optional.

But I think yeah, maybe we should bubble up the error.